packages/smooth_app/lib/pages/product/product_page.dart

```diff
@@ -72,6 +73,7 @@ class _ProductPageState extends State<ProductPage> {
 
   @override
   Widget build(BuildContext context) {
+    context.read<HistoryUpdatedSemaphore>().notify();
```

Comment:
see `_updateHistory` I think it's solving the same thing, (it's a generic model on top of localDatabase)

Reply:
Using `didChangeDependencies` as discussed.